code/modules/planet/weather.dm

```diff
@@ -0,0 +1,268 @@
+	current_weather = allowed_weather_types[new_weather]
+	next_weather_shift = world.time + rand(20, 30) MINUTES
+
+	update_icon_effects()
+	update_temperature()
+	if(old_light_modifier && current_weather.light_modifier != old_light_modifier) // Updating the sun should be done sparingly.
+		our_planet.update_sun()
+	message_admins("[our_planet.name]'s weather is now [new_weather], with a temperature of [temperature]&deg;K ([temperature - T0C]&deg;C | [temperature * 1.8 - 459.67]&deg;F).")
+
+/datum/weather_holder/proc/process()
+	if(world.time >= next_weather_shift)
+		var/new_weather
+		if(!current_weather)
+			new_weather = pickweight(roundstart_weather_chances)
+		else
+			new_weather = pickweight(current_weather.transition_chances)
+		change_weather(new_weather)
+	else
+		current_weather.process_effects()
+
+/datum/weather_holder/proc/update_icon_effects()
+	if(current_weather)
+		for(var/turf/simulated/floor/T in outdoor_turfs)
+			if(T.z in our_planet.expected_z_levels)
+				T.overlays -= T.weather_overlay
+				T.weather_overlay = image(icon = current_weather.icon, icon_state = current_weather.icon_state, layer = LIGHTING_LAYER - 1)
+				T.overlays += T.weather_overlay
+
+/datum/weather_holder/proc/update_temperature()
+	temperature = Interpolate(current_weather.temp_low, current_weather.temp_high, weight = our_planet.sun_position)
+
+	for(var/turf/unsimulated/wall/planetary/wall in planetary_walls)
+		if(ispath(wall.type, planetary_wall_type))
+			wall.temperature = temperature
+			for(var/dir in cardinal)
+				var/turf/simulated/T = get_step(wall, dir)
+				if(istype(T))
+					if(T.zone)
+						T.zone.rebuild()
+
+
+/datum/weather_holder/proc/get_weather_datum(desired_type)
+	return allowed_weather_types[desired_type]
+
+/datum/weather_holder/sif
+	temperature = T0C
+	allowed_weather_types = list(
+		WEATHER_CLEAR		= new /datum/weather/sif/clear(),
+		WEATHER_OVERCAST	= new /datum/weather/sif/overcast(),
+		WEATHER_LIGHT_SNOW	= new /datum/weather/sif/light_snow(),
+		WEATHER_SNOW		= new /datum/weather/sif/snow(),
+		WEATHER_BLIZZARD	= new /datum/weather/sif/blizzard(),
+		WEATHER_RAIN		= new /datum/weather/sif/rain(),
+		WEATHER_STORM		= new /datum/weather/sif/storm(),
+		WEATHER_HAIL		= new /datum/weather/sif/hail()
+		)
+	planetary_wall_type = /turf/unsimulated/wall/planetary/sif
+	roundstart_weather_chances = list(
+		WEATHER_CLEAR		= 30,
+		WEATHER_OVERCAST	= 30,
+		WEATHER_LIGHT_SNOW	= 20,
+		WEATHER_SNOW		= 5,
+		WEATHER_BLIZZARD	= 5,
+		WEATHER_RAIN		= 5,
+		WEATHER_STORM		= 2.5,
+		WEATHER_HAIL		= 2.5
+		)
+
+/datum/weather
+	var/name = "weather base"
+	var/icon = 'icons/effects/weather.dmi'
+	var/icon_state = null // Icon to apply to turf undergoing weather.
+	var/temp_high = T20C
+	var/temp_low = T0C
+	var/light_modifier = 1.0 // Lower numbers means more darkness.
+	var/transition_chances = list() // Assoc list
```

Comment:
The transitions happen in 10 to 15 minute intervals.  The intention with placing themselves in the probability is to act as a 'do nothing, keep raining/snowing/whatever', just so that if it snowing, there's no guarantee that it'll stop in a short period of time.